Detectors/TRD/calibration/src/DCSProcessor.cxx

```diff
@@ -163,42 +202,116 @@ int DCSProcessor::processDP(const DPCOM& dpcom)
           mLastDPTimeStamps[dpid] = etime;
         }
       }
+
+      if (std::strstr(dpid.get_alias(), "trd_fedEnvTemp") != nullptr) { // DP is trd_fedEnvTemp
+        if (!mFedEnvTempStartTSSet) {
+          mFedEnvTempStartTS = mCurrentTS;
+          mFedEnvTempStartTSSet = true;
+        }
+        auto& dpInfoFedEnvTemp = mTRDDCSFedEnvTemp[dpid];
+        if (dpInfoFedEnvTemp.nPoints == 0 || etime != mLastDPTimeStamps[dpid]) {
+          // only add data point in case last one was not already read before
+          dpInfoFedEnvTemp.addPoint(o2::dcs::getValue<double>(dpcom), etime);
+          mLastDPTimeStamps[dpid] = etime;
+        }
+      }
+
+      if (std::strstr(dpid.get_alias(), "trd_cavernHumidity") != nullptr) { // DP is trd_cavernHumidity
+        if (!mCavernStartTSSet) {
+          mCavernStartTS = mCurrentTS;
+          mCavernStartTSSet = true;
+        }
+        auto& dpInfoCavern = mTRDDCSCavern[dpid];
+        if (dpInfoCavern.nPoints == 0 || etime != mLastDPTimeStamps[dpid]) {
+          // only add data point in case last one was not already read before
+          dpInfoCavern.addPoint(o2::dcs::getValue<double>(dpcom), etime);
+          mLastDPTimeStamps[dpid] = etime;
+        }
+      }
     }
+
     if (type == DPVAL_INT) {
       if (std::strstr(dpid.get_alias(), "trd_runNo") != nullptr) { // DP is trd_runNo
         if (!mRunStartTSSet) {
           mRunStartTS = mCurrentTS;
           mRunStartTSSet = true;
         }
+
         auto& runNumber = mTRDDCSRun[dpid];
-        if (mPids[dpid] && runNumber != o2::dcs::getValue<int32_t>(dpcom)) {
-          LOGF(info, "Run number has already been processed and the new one %i differs from the old one %i", runNumber, o2::dcs::getValue<int32_t>(dpcom));
-          mShouldUpdateRun = true;
-          mRunEndTS = mCurrentTS;
-        } else {
-          runNumber = o2::dcs::getValue<int32_t>(dpcom);
-        }
-      } else if (std::strstr(dpid.get_alias(), "trd_runType") != nullptr) { // DP is trd_runType
-        if (!mRunStartTSSet) {
-          mRunStartTS = mCurrentTS;
-          mRunStartTSSet = true;
+
+        // LB: Check if new value is a valid run number (0 = cleared variable)
+        if (o2::dcs::getValue<int32_t>(dpcom) > 0) {
+          // If value has changed from previous one, new run has begun and update
+          if (o2::dcs::getValue<int32_t>(dpcom) != mCurrentRunNumber) {
+            LOG(info) << "New run number " << o2::dcs::getValue<int32_t>(dpcom) << " differs from the old one " << mCurrentRunNumber;
+            mShouldUpdateRun = true;
+            // LB: two different flags as they reset separately, after upload of CCDB, for each object
```

Comment:
why would you want to use two separate variables?

Reply:
See https://github.com/AliceO2Group/AliceO2/pull/12079#discussion_r1379732422.